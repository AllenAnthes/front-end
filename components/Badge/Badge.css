.Badge {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.iconContainer > svg {
  fill: currentColor;
  height: 100px;
  margin-bottom: 1rem;
}

.label {
  font-size: 1.2rem;
  line-height: 1.2rem;
  white-space: nowrap;
}
