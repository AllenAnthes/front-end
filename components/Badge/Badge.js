import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Badge.css';

Badge.propTypes = {
  className: PropTypes.string,

  // TODO: Create custom proptype accepting only `<svg>` or `<img>`
  // TODO: Once above is done, add integration tests regarding proptype
  icon: PropTypes.element.isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  isImageFirst: PropTypes.bool,
};

Badge.defaultProps = {
  className: undefined,
  isImageFirst: true,
};

function Badge({ className, icon, isImageFirst, label }) {
  return (
    <div className={classNames(styles.Badge, className)}>
      {isImageFirst ? (
        <>
          <div className={styles.iconContainer}>{icon}</div>
          <span className={styles.label}>{label}</span>
        </>
      ) : (
        <>
          <span className={styles.label}>{label}</span>
          <div className={styles.iconContainer}>{icon}</div>
        </>
      )}
    </div>
  );
}

export default Badge;
