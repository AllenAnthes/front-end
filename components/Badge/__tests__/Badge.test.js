/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import Icon from 'static/images/icons/github_logo.svg';

import Badge from '../Badge';

const badgeIcon = <Icon />;

describe('Badge', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Badge icon={badgeIcon} label="Testing" />);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(<Badge icon={badgeIcon} label="Badge Icon" className="test-class" />);
  });

  it('should render the image after the label when `isImageFirst` is false', () => {
    const wrapper = shallow(<Badge icon={badgeIcon} label="Badge Icon" isImageFirst={false} />);
    expect(wrapper.children().first()).toContainExactlyOneMatchingElement('span');
  });
});
