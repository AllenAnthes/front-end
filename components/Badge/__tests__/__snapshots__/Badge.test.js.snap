// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render with many props assigned 1`] = `
<div
  className="Badge test-class"
>
  <div
    className="iconContainer"
  >
    <svg />
  </div>
  <span
    className="label"
  >
    Badge Icon
  </span>
</div>
`;

exports[`Badge should render with required props 1`] = `
<div
  className="Badge"
>
  <div
    className="iconContainer"
  >
    <svg />
  </div>
  <span
    className="label"
  >
    Testing
  </span>
</div>
`;
